zips/zip-0246.rst

```diff
@@ -32,41 +32,58 @@ field, as specified in section 7.1 of the Zcash protocol specification
 Abstract
 ========
 
-This ZIP defines the digest algorithms associated with the v6 transaction format.
-
-This proposal also defines the new concept of "sighash versioning": where previously
-each transaction version had a single associated digest algorithm, going forward it
-will be possible for signers to use any digest algorithm within the closed set
-specified for a given transaction version (and made available in consensus via network
-upgrades).
+This ZIP defines the digest algorithms associated with the v6 transaction
+format.
 
+This proposal also defines the new concept of "sighash algorithm versioning":
+where previously each transaction version had a single associated digest
+algorithm, going forward it will be possible for signers to use any digest
+algorithm within the closed set specified for a given transaction version (and
+made available in consensus via network upgrades).
 
 ==========
 Motivation
 ==========
 
-TBD
-
-TODO: Motivate sighash versioning
+The motivation for sighash algorithm versioning is that we want to be able to
+specify the V6 transaction format and signatures without knowing what the
+signatures for atomic swaps in action bundles will need to commit to. This
+allows for future signature hash algorithms to be specified for V6 transactions
+that support signing parts of a transaction while leaving the remainder of the
+transaction malleable, or that commit to additional state that nodes may start
+tracking in a future upgrade that is not part of an individual transaction
+being signed.
 
+Being able to remove support for sighash algorithm versions makes it possible
+to respond effectively in the case that a bug is found in a previously-deployed
```

Comment:
```suggestion
to respond effectively in the case that a bug is found in a previously deployed
```
"previously deployed" is an adjective modified by an adverb, not a compound adjective.